### PR TITLE
LWM2M device level logging event configuration

### PR DESCRIPTION
**Pull request by JacekGonera**
Task: https://cumulocity.atlassian.net/browse/DM-4190

### CODE REVIEW

File: content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md [hunk 1]
```diff
@@ -433,6 +433,16 @@ The value must not exceed the maximum request timeout limit given in the LWM2M m
...
 <td style="text-align: left">Optional</td>
 </tr>
+<tr>
+<td style="text-align: left">Event log level</td>
+<td style="text-align: left">logLevel</td>
+<td style="text-align: left">String</td>
+<td style="text-align: left">{{< product-c8y-iot >}} can output detailed logs to event stream. This field configures a log level. Allowed values are: 
+<b>NONE</b> (nothing will be logged as events), <b>LIFECYCLE</b> (only registration, de-registration and registration update events),
+<b>TRAFFIC</b> (LIFECYCLE + sent/received data), <b>FIRMWARE</b> (LIFECYCLE + detailed firmware update information),
+<b>VERBOSE</b> (all above will be logged)</td>
```
Comment (Oezge-Akin) [inline content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md]:
```suggestion
<b>VERBOSE</b> (all above will be logged).
Default is LIFECYCLE.</td>
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+product_area: Device management & connectivity
+change_type:
+  - value: change-QHu1GdukP
+    label: Feature
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ticket: DM-4190
+version: 1020.438.0
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
version: 10.20.438.0
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+---
+For LWM2M devices, it is now possible to adjust logging event level. It can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. For more details see [LWM2M configuration](/protocol-integration/lwm2m/#lwm2m-configuration).
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
For LWM2M devices, it is now possible to adjust or disable logging event level per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
For more details see [LWM2M device registration advanced settings](/protocol-integration/lwm2m/#advanced-settings) and [LWM2M configuration](/protocol-integration/lwm2m/#lwm2m-configuration). 
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+date: 2024-12-12
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
date: 2024-12-19
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+---
+date: 2024-12-19
+title: LWM2M device level logging event configuration
```
Comment (aaronraab) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
title: LWM2M device logging level configuration
```

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-ggH2M4hf3
+    label: lwm2m-agent
+ticket: DM-4190
+version: 10.20.438.0
+---
+For LWM2M devices, it is now possible to adjust or disable logging event level per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
```
Comment (aaronraab) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
```

Comment (aaronraab) [reply content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]: but wouldn't a customer actually prefer or expect to define the default as well? I guess most want to change the defualt to disabled and not have to go through each device.

Comment (Oezge-Akin) [reply content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]: @aaronraab Lifecycle events always got created. We didn't want to introduce the behavior change. Setting the tenant level defaults can be implemented later but currently we don't have such feature.

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 8]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * "NONE" - nothing will be logged as events,
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * NONE - nothing will be logged as events,
```

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 9]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * NONE - nothing will be logged as events,
+    * "LIFECYCLE" - only registration, de-registration and registration update events,
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * LIFECYCLE - only registration, de-registration and registration update events
```

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 10]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * NONE - nothing will be logged as events,
+    * LIFECYCLE - only registration, de-registration and registration update events
+    * "TRAFFIC" - same as LIFECYCLE + sent/received data,
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * TRAFFIC - same as LIFECYCLE plus sent/received data
```

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 11]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * NONE - nothing will be logged as events,
+    * LIFECYCLE - only registration, de-registration and registration update events
+    * TRAFFIC - same as LIFECYCLE plus sent/received data
+    * "FIRMWARE" - same as LIFECYCLE + detailed firmware update information,
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * FIRMWARE - same as LIFECYCLE plus detailed firmware update information
```

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 12]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * NONE - nothing will be logged as events,
+    * LIFECYCLE - only registration, de-registration and registration update events
+    * TRAFFIC - same as LIFECYCLE plus sent/received data
+    * FIRMWARE - same as LIFECYCLE plus detailed firmware update information
+    * "VERBOSE" - all above details will be logged
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * VERBOSE - all above details will be logged
```

File: content/protocol-integration/lwm2m-bundle/device-details.md [hunk 13]
```diff
@@ -92,6 +92,12 @@ The **LWM2M configuration** tab displays all LWM2M settings related to the devic
   * Delivery protocol
   * Reset state machine settings
   * Fail device firmware update on unexpected result (terminate the device firmware update process if an unexpected situation is detected)
+  * Logging event level - the logging levels can be adjusted based on types or completely be disabled for the device. When enabled, log events are visible in the device's events. Levels:
+    * NONE - nothing will be logged as events,
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/device-details.md]:
```suggestion
    * NONE - nothing will be logged as events
```

File: content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md [hunk 14]
```diff
@@ -433,6 +433,17 @@ The value must not exceed the maximum request timeout limit given in the LWM2M m
 <td style="text-align: left">{{< product-c8y-iot >}} implements default handlers for objects 3,4 and 6. For example, they update the device name upon the reception of the corresponding resource in object 3 or update the device location. This flag allows those behaviors to be turned off.</td>
 <td style="text-align: left">Optional</td>
 </tr>
+<tr>
+<td style="text-align: left">Event log level</td>
+<td style="text-align: left">logLevel</td>
+<td style="text-align: left">String</td>
+<td style="text-align: left">{{< product-c8y-iot >}} can output detailed logs to event stream. This field configures a log level. Allowed values are:
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md]:
```suggestion
<td style="text-align: left">{{< product-c8y-iot >}} can output detailed logs to the event stream. This field configures a log level. Allowed values are: 
```

File: content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md [hunk 15]
```diff
@@ -433,6 +433,17 @@ The value must not exceed the maximum request timeout limit given in the LWM2M m
...
 <td style="text-align: left">Optional</td>
 </tr>
+<tr>
+<td style="text-align: left">Event log level</td>
+<td style="text-align: left">logLevel</td>
+<td style="text-align: left">String</td>
+<td style="text-align: left">{{< product-c8y-iot >}} can output detailed logs to the event stream. This field configures a log level. Allowed values are: 
+<b>NONE</b> (nothing will be logged as events), <b>LIFECYCLE</b> (only registration, de-registration and registration update events),
+<b>TRAFFIC</b> (LIFECYCLE + sent/received data), <b>FIRMWARE</b> (LIFECYCLE + detailed firmware update information),
+<b>VERBOSE</b> (all above will be logged).
```
Comment (BeateRixen) [inline content/protocol-integration/lwm2m-bundle/registering-lwm2m-devices.md]:
```suggestion
<b>VERBOSE</b> (all of the above will be logged).
```

**Review by Oezge-Akin — APPROVED** [review]

File: content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-ggH2M4hf3
+    label: lwm2m-agent
+ticket: DM-4190
+version: 10.20.438.0
+---
+For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from LWM2M Configuration tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from LWM2M Service to enabling event logging is removed.
```
Comment (BeateRixen) [inline content/change-logs/device-management/ui-c8y-1020-438-0-LWM2M-device-level-logging-event-configuration.md]:
```suggestion
For LWM2M devices, it is now possible to disable or adjust the granularity of logging events per device during the device registration or from the **LWM2M configuration** tab of the existing device. When enabled, log events are visible in the device's events. Lifecycle events are the default selected option. The global configuration from the LWM2M Service for enabling event logging has been removed.
```

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**